src/Metadata.php

```diff
@@ -169,4 +175,61 @@ protected function getPlaceholderIllustration()
     {
         return 'app/img/logo.png';
     }
+
+    /**
+     * @param $file
+     *
+     * @return array
+     */
+    protected function getEntriesFromCSV($file)
+    {
+        return Reader::createFromPath($file)->fetchAssoc(0);
+    }
+
+    /**
+     * @param $file
+     *
+     * @return array
+     */
+    protected function getEntriesFromCache($file)
+    {
+        $identifier = $this->getCacheIdentifier($file);
+        $lastModifiedAt = (new SplFileInfo($file))->getMTime();
+
+        if ($cached = $this->cache->get($identifier)) {
+            if (!$this->isModified($cached, $lastModifiedAt)) {
+                return $cached['meta'];
+            }
+        };
+
+        $entries = $this->getEntriesFromCSV($file);
+
+        $this->cache->forever($identifier, [
+            'last_modified_at' => $lastModifiedAt,
+            'meta' => $entries
+        ]);
```

Comment:
It's indeed shorter yes but it also means that the entries will remain in the cache forever (till the next `cache:clear`). With this approach there is only ever going to be one set of meta data for the given file.

I know that this might seem a bit radical, but if you have an application that heavily relies on a cache you can't afford to

*. Keep unused stuff in the cache (since it will take space)
*. Run `artisan cache:clear` each time you deploy